src/Microsoft.ML/Runtime/Internal/Tools/GeneratorUtils.cs

```diff
@@ -0,0 +1,467 @@
+﻿using Microsoft.CSharp;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Newtonsoft.Json.Linq;
+using System;
+using System.CodeDom;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.ML.Runtime.Internal.Tools
+{
+    internal static class GeneratorUtils
+    {
+        public static string GetFullMethodName(ModuleCatalog.EntryPointInfo entryPointInfo)
+        {
+            return entryPointInfo.Name;
+        }
+
+        public class EntryPointGenerationMetadata
+        {
+            public string Namespace { get; private set; }
+            public string ClassName { get; private set; }
+            public EntryPointGenerationMetadata(string @namespace, string className)
+            {
+                Namespace = @namespace;
+                ClassName = className;
+            }
+        }
+
+        public static EntryPointGenerationMetadata GetEntryPointMetadata(ModuleCatalog.EntryPointInfo entryPointInfo)
+        {
+            var split = entryPointInfo.Name.Split('.');
+            Contracts.Assert(split.Length == 2);
```

Comment:
>Contracts.Assert(split.Length == 2); [](start = 12, length = 36)

An assert is not appropriate, because certainly it is possible for a component to be written with other than exactly one period.

Reply:
All our current code with entry points assume what name looks like {namespace}.{method}  for example: Transforms.LabelColumnKeyBooleanConverter or Trainers.LogisticRegressionClassifier or Models.RegressionEvaluator.
I'm honestly not sure how we enforce it, but looks like we somehow enforce it.

---
In reply to: [194211131](https://github.com/dotnet/machinelearning/pull/339#discussion_r194211131) [](ancestors = 194211131)

Reply:
If you aren't certain that it will necessarily be the case, then we ought to make this a check.

---
In reply to: [194556586](https://github.com/dotnet/machinelearning/pull/339#discussion_r194556586) [](ancestors = 194556586,194211131)